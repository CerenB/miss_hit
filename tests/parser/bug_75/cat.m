% (c) Copyright 2020 Florian Schanda
classdef cat < potato.kitten
    methods
        function meow(self)
            % calls superclass meow, but uses a dotted name
            self.meow@potato.kitten;
            disp("*destory stuff*");
        end
    end
end
