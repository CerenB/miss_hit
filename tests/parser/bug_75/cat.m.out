----------------------------------------------------------------------
--  Parse tree for cat.m
Root: Class_File
  Classdef: Class_Definition
    Name: Identifier <cat>
    Superclasses: Selection
      Prefix: Identifier <potato>
      Field: Identifier <kitten>
    Methods: Methods Special_Block
      Items: Function_Definition
        Signature: Function_Signature
          Name: Identifier <meow>
          Inputs: Identifier <self>
        Body: Sequence_Of_Statements
          Statements: Naked_Expression_Statement
            Expression: Superclass_Reference
              Prefix: Selection
                Prefix: Identifier <self>
                Field: Identifier <meow>
              Reference: Selection
                Prefix: Identifier <potato>
                Field: Identifier <kitten>
          Statements: Naked_Expression_Statement
            Expression: Reference
              Name: Identifier <disp>
              Arguments: String_Literal "*destory stuff*"
----------------------------------------------------------------------
In cat.m, line 2
| classdef cat < potato.kitten
|              ^ info: this token is not linked to the ast
In cat.m, line 4
|         function meow(self)
|                      ^ info: this token is not linked to the ast
In cat.m, line 4
|         function meow(self)
|                           ^ info: this token is not linked to the ast
In cat.m, line 6
|             self.meow@potato.kitten;
|                                    ^ info: this token is not linked to the ast
In cat.m, line 7
|             disp("*destory stuff*");
|                 ^ info: this token is not linked to the ast
In cat.m, line 7
|             disp("*destory stuff*");
|                                   ^ info: this token is not linked to the ast
In cat.m, line 7
|             disp("*destory stuff*");
|                                    ^ info: this token is not linked to the ast
In cat.m, line 8
|         end
|         ^^^ info: this token is not linked to the ast
In cat.m, line 9
|     end
|     ^^^ info: this token is not linked to the ast
In cat.m, line 10
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
